### ISSUE TITLE
Call for unimim's code

### ISSUE THREAD

**Issue by jsrdcht** [microsoft/unilm#1084]
**The flowers I was waiting for have all withered.**

**Comment by pengzhiliang** [comment]
I apologize for the delay. I will release the code as soon as possible. Please stay tuned, and thank you for your support and understanding!

**Comment by jsrdcht** [comment]
> I apologize for the delay. I will release the code as soon as possible. Please stay tuned, and thank you for your support and understanding!

Hope to release the code as soon as possible. I can't reproduce the accuracy of the article.

**Comment by drkostas** [comment]
**tl;dr** I implemented MaskDistill from scratch in PyTorch, reproduced the paper's experiments, and open sourced the code and weights: https://github.com/drkostas/MaskDistill-PyTorch

I needed this for my research and the existing unofficial implementation (https://github.com/bwconrad/masked-distillation) was not giving me the expected results, so I ended up writing everything from scratch. ViT-Base pretrained for 300 epochs with CLIP teacher reproduces:

- Finetuning: 84.8% top-1 (paper reports 85.3%)
- Linear probe: 76.3% top-1
- Semantic segmentation: 52.6 mIoU (paper reports 53.8)
- k-NN: 75.6%
- Object detection: 44.4 bbox mAP
- Instance segmentation: 40.1 segm mAP

All checkpoints on HuggingFace: https://huggingface.co/drkostas/MaskDistill-ViT-Base
